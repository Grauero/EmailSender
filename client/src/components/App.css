.container {
  min-width: '520px';
}

.modal {
  z-index: 10000;
  background-color: 'rgba(0, 0, 0, 0.5)';
  position: 'fixed';
  top: 0;
  left: 0;
  width: '100%';
  height: '100vh';
  text-align: 'center';
}

.message {
  z-index: 10000;
  background-color: 'rgba(255, 255, 255, .9)';
  width: '50%';
  text-align: 'center';
  display: 'inline-block';
  margin: '15em auto';
  margin-left: '20%';
  margin-right: '20%';
  border-radius: '30px';
}

.paragrapgh {
  z-index: 10000;
  padding-top: '10px';
  font-size: '1.5em';
}

.button {
  z-index: 10000;
  margin: '3em 5%';
}

.credits {
  margin: '0 10px';
}

.link {
  margin-left: '10px';
}

.input {
  margin-bottom: '5px';
}

.touched {
  margin-bottom: '20px';
}

.survey-card {
  position: 'relative';
}

.close {
  position: 'absolute';
  right: 0;
  cursor: 'pointer';
}

.response {
  font-size: '1.5em';
}

.positive-response {
  letter-spacing: '2px';
  margin-top: '10px';
  text-align: 'center';
}

.negative-response {
  letter-spacing: '2px';
  margin-top: '10px';
  text-align: 'center';
}
