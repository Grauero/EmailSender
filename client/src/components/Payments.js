import React from 'react';
import PropTypes from 'prop-types';
import StripeCheckout from 'react-stripe-checkout';
import { connect } from 'react-redux';

import { handleToken } from '../store/actions';

const Payments = ({ handleToken }) => (
  <StripeCheckout
    name="EmailSender"
    description="$5 for 5 email credits"
    amount={500}
    token={token => handleToken(token)}
    stripeKey={process.env.REACT_APP_STRIPE_KEY}
  >
    <button type="button" className="btn">
      Add Credits
    </button>
  </StripeCheckout>
);

Payments.propTypes = {
  handleToken: PropTypes.func.isRequired
};

export default connect(
  null,
  { handleToken }
)(Payments);
