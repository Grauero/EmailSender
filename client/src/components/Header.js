import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Payments from './Payments';

const Header = ({ auth }) => {
  function renderContent() {
    switch (auth) {
      case null:
        return;
      case false:
        return (
          <li>
            <a href={process.env.REACT_APP_LOGIN}>Login With Google</a>
          </li>
        );
      default:
        return [
          <li key="1">
            <Payments />
          </li>,
          <li key="2" style={{ margin: '0 10px' }}>
            Credits: {auth.credits}
          </li>,
          <li key="3">
            <a href={process.env.REACT_APP_LOGOUT}>Logout</a>
          </li>
        ];
    }
  }

  return (
    <nav>
      <div className="nav-wrapper">
        <Link
          to={auth ? '/surveys' : '/'}
          className="left brand-logo"
          style={{ marginLeft: '10px' }}
        >
          <i className="material-icons left">email</i>EmailSender
        </Link>
        <ul className="right">{renderContent()}</ul>
      </div>
    </nav>
  );
};

Header.propTypes = {
  auth: PropTypes.oneOfType([PropTypes.instanceOf(Object), PropTypes.bool])
};

Header.defaultProps = {
  auth: null
};

const mapStateToProps = state => ({
  auth: state.auth
});

export default connect(mapStateToProps)(Header);
