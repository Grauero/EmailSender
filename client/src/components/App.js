import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import 'materialize-css/dist/css/materialize.min.css';

import store from '../store/store';
import Header from './Header';

const Dashboard = () => <h2>Dashboard</h2>;
const SurveyNew = () => <h2>SurveyNew</h2>;
const Landing = () => <h2>Landing</h2>;

const App = () => (
  <Provider store={store}>
    <div className="container">
      <BrowserRouter>
        <div>
          <Header />
          <Route exact path="/" component={Landing} />
          <Route exact path="/surveys" component={Dashboard} />
          <Route exact path="/surveys/new" component={SurveyNew} />
        </div>
      </BrowserRouter>
    </div>
  </Provider>
);

export default App;
