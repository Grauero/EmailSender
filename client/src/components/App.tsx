import React, { Component, lazy, Suspense } from 'react';
import { Dispatch, bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { BrowserRouter, Route } from 'react-router-dom';
import { fetchUser } from '../store/actions';
import 'materialize-css/dist/css/materialize.min.css';

import Header from './nav/Header';
import Landing from './nav/Landing';
import Spinner from './utils/Spinner';
import './App.css';

const Dashboard = lazy(() => import('./dashboard/Dashboard'));
const SurveyNew = lazy(() => import('./surveys/SurveyNew'));

class App extends Component<ReturnType<typeof mapDispatchToProps>> {
  public componentDidMount() {
    this.props.fetchUser();
  }

  public render() {
    return (
      <BrowserRouter>
        <Suspense fallback={<Spinner />}>
          <Header />
          <div className="dashboard">
            <Route exact path="/" component={Landing} />
            <Route exact path="/surveys" component={Dashboard} />
            <Route exact path="/surveys/new" component={SurveyNew} />
          </div>
        </Suspense>
      </BrowserRouter>
    );
  }
}

const mapDispatchToProps = (dispatch: Dispatch) => {
  return bindActionCreators({ fetchUser }, dispatch);
};

export default connect(
  null,
  mapDispatchToProps
)(App);
