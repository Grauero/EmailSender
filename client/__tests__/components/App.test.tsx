import 'jest';
import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import { App, mapDispatchToProps } from '../../src/components/App';

const props = { fetchUser: jest.fn() };
const component = shallow(<App {...props} />);

test('calls props.fetchUser when component mounts', () => {
  expect(props.fetchUser).toHaveBeenCalled();
  expect(props.fetchUser).toHaveBeenCalledTimes(1);
});

test('matches snapshot', () => expect(toJSON(component)).toMatchSnapshot());

test('dispatches fetchUser from store to props', () => {
  const props = mapDispatchToProps(jest.fn());

  expect(Object.keys(props)[0]).toBe('fetchUser');
});
