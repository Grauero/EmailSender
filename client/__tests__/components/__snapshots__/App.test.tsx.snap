// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<BrowserRouter>
  <React.Suspense
    fallback={<Spinner />}
  >
    <Connect(Header) />
    <div
      className="dashboard"
    >
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        exact={true}
        path="/surveys"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.lazy),
            "_ctor": [Function],
            "_result": null,
            "_status": -1,
          }
        }
        exact={true}
        path="/surveys/new"
      />
    </div>
  </React.Suspense>
</BrowserRouter>
`;
